userspace/libsinsp/filterchecks.cpp

```diff
@@ -4777,6 +4777,7 @@ uint8_t* sinsp_filter_check_user::extract(sinsp_evt *evt, OUT uint32_t* len, boo
 	case TYPE_LOGINNAME:
 		ASSERT(m_inspector != NULL);
 		uinfo = m_inspector->get_user(tinfo->m_loginuid);
+		ASSERT(uinfo != NULL);
```

Comment:
Same as for the group name. Not having a name for a uid is maybe unexpected but certainly not a crash-worthy error. So the ASSERT in line 4758 is wrong too IMHO.

Also, loginuid can easily be -1.